portal/import_template_ui.php

```diff
@@ -15,303 +15,543 @@
-$tdir = $GLOBALS['OE_SITE_DIR'] . '/documents/onsite_portal_documents/templates/';
-if (!empty($patient_dir)) {
-    $tdir = $GLOBALS['OE_SITE_DIR'] . '/documents/onsite_portal_documents/templates/' . $patient_dir . '/';
-} elseif (!empty($cat_dir)) {
-    $tdir = $GLOBALS['OE_SITE_DIR'] . '/documents/onsite_portal_documents/templates/' . $cat_dir . '/';
-}
+$templateService = new DocumentTemplateService();
 
-$rtn = sqlStatement("SELECT `option_id`, `title`, `seq` FROM `list_options` WHERE `list_id` = ? ORDER BY `seq`", array('Document_Template_Categories'));
-$category_list = array();
-while ($row = sqlFetchArray($rtn)) {
-    $category_list[] = $row;
-}
+$patient = $_REQUEST['selected_patients'] ?? null;
+$patient = $patient ?: ($_REQUEST['upload_pid'] ?? 0);
 
-function getAuthUsers()
-{
-    $response = sqlStatement("SELECT patient_data.pid, Concat_Ws(' ', patient_data.fname, patient_data.lname) as ptname FROM patient_data WHERE allow_patient_portal = 'YES'");
-    $resultpd = array();
-    while ($row = sqlFetchArray($response)) {
-        $resultpd[] = $row;
-    }
+$category = $_REQUEST['template_category'] ?? '';
+$category_list = $templateService->getDefaultCategories();
 
-    return $resultpd;
-}
+$none_message = xlt("Nothing to show for current actions.");
 
-function getTemplateList($dir, $location = "")
+function getAuthUsers()
 {
-    $retval = array();
-    if (substr($dir, -1) !== "/") {
-        $dir .= "/";
-    }
-
-    if (false === $d = @dir($dir)) {
-        return false;
-    }
-    while (false !== ($entry = $d->read())) {
-        if ($entry[0] === "." || substr($entry, -3) !== 'tpl') {
-            continue;
-        }
-        if (is_dir("$dir$entry")) {
-            continue;
-        }
+    $response = sqlStatement("SELECT `pid`, Concat_Ws(', ', `lname`, `fname`) as ptname FROM `patient_data` WHERE `allow_patient_portal` = 'YES' ORDER BY `lname`");
 
-        if (is_readable("$dir$entry")) {
-            $retval[] = array(
-                'pathname' => "$dir$entry",
-                'name' => "$entry",
-                'size' => filesize("$dir$entry"),
-                'lastmod' => filemtime("$dir$entry"),
-                'location' => text("./documents/onsite_portal_documents/templates/" . $location)
-            );
-        }
+    $result_data = array(
+        ['pid' => '0', 'ptname' => 'All Patients'],
+        ['pid' => '-1', 'ptname' => 'Repository'],
+    );
+    while ($row = sqlFetchArray($response)) {
+        $result_data[] = $row;
     }
 
-    $d->close();
-    return $retval;
+    return $result_data;
 }
 
 ?>
 <!DOCTYPE html>
-<html>
 <head>
     <meta charset="UTF-8">
     <title><?php echo xlt('Portal'); ?> | <?php echo xlt('Templates'); ?></title>
     <meta name="description" content="Developed By sjpadgett@gmail.com">
-    <?php Header::setupHeader(['datetime-picker', 'summernote', 'summernote-ext-nugget']); ?>
+    <?php Header::setupHeader(['datetime-picker', 'select2', 'ckeditor']); ?>
```

Comment:
wooohooooo! does this mean we can get rid of these dependencies?
(if so, i'd be happy to do this after this PR comes in)

Reply:
Nope I still have to refactor secure messaging and one other I forget which. But soon